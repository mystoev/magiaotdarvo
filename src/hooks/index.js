export { useCategories } from './use-categories';
export { useToken } from './use-login';
export { useProduct } from './use-product';
export { useProducts } from './use-products';
export { useScrollTop } from './use-scroll-top';
export { useSendEmail } from './use-send-email';
export { useUpdateData, useAddData, useUploadImages } from './use-update-data';
