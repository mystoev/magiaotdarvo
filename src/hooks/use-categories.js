import { useQuery } from 'react-query';
import axios from 'axios';

import { IS_PROD, LOCAL_SERVER } from '../constants';

const CATEGORIES_PHP = 'categories.php';
const CATEGORIES_KEY = 'categories';

const getCategories = async () => {
  const endpoint = IS_PROD ? CATEGORIES_PHP : LOCAL_SERVER + CATEGORIES_PHP;
  const { data } = await axios.get(endpoint);

  return JSON.parse(data);
};

export const useCategories = () => useQuery(CATEGORIES_KEY, () => getCategories());
