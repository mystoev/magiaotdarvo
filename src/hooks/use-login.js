import { useState } from 'react';
import axios from 'axios';

const LOGIN_FILE = 'login.php';

export const login = async (user, password) => {
  try {
    const { data } = await axios.post(LOGIN_FILE, { user, password });
    if (data) {
      localStorage.setItem('token', data);
      return data;
    }
  } catch {
    return null;
  }
};

export const useToken = () => {
  const [token, setToken] = useState(localStorage.getItem('token'));

  return [token, setToken];
};
