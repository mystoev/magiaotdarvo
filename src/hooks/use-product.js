import { useQuery } from 'react-query';
import axios from 'axios';

import { IS_PROD, PRODUCT_PHP, LOCAL_SERVER } from '../constants/data';

const getProduct = async ({ categoryName, productName }) => {
  const endpoint = IS_PROD ? PRODUCT_PHP : LOCAL_SERVER + PRODUCT_PHP;
  const { data } = await axios.get(endpoint + `?category=${categoryName}&product=${productName}`);

  return JSON.parse(data);
};

export const useProduct = ({ categoryName, productName }) =>
  useQuery(`${categoryName}/${productName}`, () => getProduct({ categoryName, productName }));
