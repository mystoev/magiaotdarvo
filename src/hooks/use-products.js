import { useQuery } from 'react-query';
import axios from 'axios';

import { IS_PROD, PRODUCTS_PHP, LOCAL_SERVER } from '../constants/data';

const getProducts = async (categoryName) => {
  const endpoint = IS_PROD ? PRODUCTS_PHP : LOCAL_SERVER + PRODUCTS_PHP;
  const { data } = await axios.get(endpoint + `?category=${categoryName}`);

  return JSON.parse(data);
};

export const useProducts = ({ categoryName }) =>
  useQuery(categoryName, () => getProducts(categoryName));
