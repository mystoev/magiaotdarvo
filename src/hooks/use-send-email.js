import { useMutation } from 'react-query';
import axios from 'axios';

import { EMAIL_FILE } from '../constants/data';
import { selectToken } from '../selectors/token';

const sendEmail = async ({ name, email, phone, message, productLink }) => {
  const token = selectToken();
  const payload = {
    key: token,
    subject: `Запитване от ${name}`,
    message: `
      --------
      Имейл: ${email}
      Телефон: ${phone} ${productLink ? `\n      Относно: ${productLink}` : ''}
      --------
      ${message}
    `
  };

  try {
    const { data } = await axios.post(EMAIL_FILE, payload);
    console.log(data);
  } catch (error) {
    console.log('ERROR SENDING EMAIL');
    console.log(error);
  }
};

export const useSendEmail = (options) => useMutation(sendEmail, options);
