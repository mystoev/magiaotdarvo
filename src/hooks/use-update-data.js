import { UPDATE_FILE, ADD_FILE, UPLOAD_FILE, IS_PROD, LOCAL_SERVER } from '../constants';
import axios from 'axios';
import { selectToken } from '../selectors/token';

export const useUpdateData = async ({
  category,
  productName,
  newProductName,
  description,
  imagesColumn
}) => {
  const token = selectToken();
  const payload = {
    key: token,
    category,
    productName,
    newProductName,
    description,
    imagesColumn
  };

  const endpoint = IS_PROD ? UPDATE_FILE : LOCAL_SERVER + UPDATE_FILE;
  const { data } = await axios.post(endpoint, payload);
  return data;
};

export const useAddData = async ({ category, newProductName, description, images }) => {
  const token = selectToken();
  const payload = {
    key: token,
    category,
    newProductName,
    description,
    images
  };

  const endpoint = IS_PROD ? ADD_FILE : LOCAL_SERVER + ADD_FILE;
  const { data } = await axios.post(endpoint, payload);
  return data;
};

export const useUploadImages = async ({ category, productName, images }) => {
  const token = selectToken();
  const payload = {
    key: token,
    category,
    productName,
    images
  };

  const endpoint = IS_PROD ? UPLOAD_FILE : LOCAL_SERVER + UPLOAD_FILE;
  const { data } = await axios.post(endpoint, payload);
  return data;
};
