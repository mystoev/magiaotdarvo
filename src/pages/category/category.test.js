import React from 'react';
import { fireEvent } from '@testing-library/react';
import { renderWithTheme } from '../../../test/utils';
import { MemoryRouter as Router } from 'react-router-dom';

import { useProducts } from '../../hooks/use-products';
import Category, { ITEMS_PER_PAGE } from './';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: jest.fn().mockReturnValue({ id: 'test-product' })
}));
jest.mock('../../hooks/use-products', () => ({
  useProducts: jest.fn().mockReturnValue({
    data: [
      {
        folder: 'test-folder',
        name: 'test-name',
        files: ['test-file1'],
        category: 'test-category',
        images: ''
      }
    ]
  })
}));
jest.mock('../../components/page-header', () => ({ title, description }) => (
  <div>
    {title} {description}
  </div>
));

describe('category page', () => {
  it('should render', () => {
    const { container } = renderWithTheme(
      <Router>
        <Category />
      </Router>
    );

    expect(container).toMatchSnapshot();
  });

  it('should handle page change', () => {
    const products = [];
    for (let index = 0; index < ITEMS_PER_PAGE; index++) {
      products.push({
        folder: 'test-folder',
        name: 'test-name',
        files: ['test-file1'],
        category: 'test-category',
        images: 'cover.jpg,test_image.jpg'
      });
    }

    products.push({
      folder: 'test-folder',
      name: 'test-name-2',
      files: ['test-file1'],
      category: 'test-category',
      images: ''
    });

    useProducts.mockReturnValue({
      data: products
    });
    const { queryByText } = renderWithTheme(
      <Router>
        <Category />
      </Router>
    );

    expect(queryByText('test-name-2')).not.toBeInTheDocument();

    fireEvent.click(queryByText('2'));
    expect(queryByText('test-name-2')).toBeInTheDocument();
  });

  it('should change pages', () => {
    useProducts.mockReturnValue({
      data: null
    });

    renderWithTheme(
      <Router>
        <Category />
      </Router>
    );
  });
});
