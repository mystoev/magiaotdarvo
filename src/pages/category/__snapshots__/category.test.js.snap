// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`category page should render 1`] = `
.c1 {
  width: 100%;
  background-color: #EEC61F;
  opacity: 0.5;
}

.c2 {
  height: 1px;
}

.c0 {
  width: 95%;
  margin: 20px auto;
  display: flex;
  flex-wrap: wrap;
  justify-content: normal;
  column-gap: 2%;
  row-gap: 20px;
}

.c0 >.category-link {
  text-align: center;
  flex-basis: 23%;
  box-shadow: 0px 0px 2px #1C1311;
}

.c0 >.category-link .product-title-container {
  flex-grow: 1;
  display: flex;
  flex-direction: column;
  justify-content: center;
  background: #1C1311;
}

.c0 >.category-link a {
  height: 100%;
  display: flex;
  flex-direction: column;
  color: #EEC61F;
}

.c0 >.category-link .product-image-container {
  position: relative;
  height: 200px;
}

.c0 >.category-link img {
  position: absolute;
  left: 0;
  width: 100%;
  height: 100%;
  object-fit: contain;
}

.c0 >.category-link img.image-cover {
  opacity: 0.75;
  object-fit: fill;
  filter: blur(5px);
  overflow: hidden;
}

.c0 >.category-link .product-title,
.c0 >.category-link .product-description {
  white-space: break-spaces;
  word-wrap: break-word;
}

.c0 >.category-link .product-title {
  padding: 5px 0px;
}

.c0 >.category-link .product-description {
  padding-bottom: 5px;
  font-size: 0.7em;
}

@media only screen and (max-width:768px) {
  .c0 >.category-link {
    flex-basis: 90%;
  }
}

<div>
  <div>
     
  </div>
  <div
    class="c0"
  >
    <div
      class="category-link"
    >
      <a
        href="/product/test-product/test-folder"
      >
        <div
          class="product-image-container"
        >
          <img
            class="image-cover"
            src=""
          />
          <img
            src=""
          />
        </div>
        <div
          class="c1 c2"
        />
        <div
          class="product-title-container"
        >
          <p
            class="product-title"
          >
            test-name
          </p>
        </div>
        <div
          class="c1 c2"
        />
      </a>
    </div>
  </div>
</div>
`;
